### PR TITLE
Enhance webpack experience

### PR DESCRIPTION
**Pull request by racse1**
Build times are now 2.5 times faster and output file may have a smaller size.

### CODE REVIEW

File: app/templates/_webpack.config.js [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+const production   = process.env.NODE_ENV === 'production';
 
 module.exports = {
     entry: {
         '<%= answers.projectName %>': [conf.js.src + '/<%= answers.projectName %>/index.js']
     },
     output: {
-        path: path.resolve(__dirname, conf.js.dest + '/'),
+        path: path.resolve(__dirname, conf.js.dest),
+        publicPath: conf.js.dest.replace(conf.baseDir, '') + '/',
```
Comment (racse1) [inline app/templates/_webpack.config.js]: Removed hardcoded `publicPath`.

File: app/templates/_webpack.config.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
         filename: '[name].bundle.js',
-        chunkFilename: '[name].[chunkhash].js',
-        publicPath: '/Content/Scripts/'
+        chunkFilename: '[name].[chunkhash].chunk.js'
     },
-    devtool: 'source-maps',
+    devtool: production ? 'source-maps' : 'cheap-module-eval-source-map',
```
Comment (racse1) [inline app/templates/_webpack.config.js]: Faster source map for development.

File: app/templates/_webpack.config.js [hunk 3]
```diff
@@ -23,16 +24,16 @@ module.exports = {
         ]
     },
     plugins: [
+        new webpack.DefinePlugin({
+            'process.env.NODE_ENV': production ? '"production"' : '"development"'
```
Comment (racse1) [inline app/templates/_webpack.config.js]: Some libraries (Vue.js, React) do use `process.env.NODE_ENV` property in their code. We need to set it to `"production"` in order to make sure some redundant debug code is removed from our bundles.

File: app/templates/_webpack.config.js [hunk 4]
```diff
@@ -23,16 +24,16 @@ module.exports = {
...
+        }),
         new webpack.ProvidePlugin({
             'window.jQuery': 'jquery',
             'window.$': 'jquery'
         }),
-        new webpack.optimize.UglifyJsPlugin({
-            compress: {
-                warnings: false
```
Comment (racse1) [inline app/templates/_webpack.config.js]: Warnings are `false` [by default](https://github.com/webpack-contrib/uglifyjs-webpack-plugin/tree/master#uglifyoptions).

File: app/templates/_webpack.config.js [hunk 5]
```diff
@@ -23,16 +24,16 @@ module.exports = {
...
-            },
+        ...production ? [new webpack.optimize.UglifyJsPlugin({
```
Comment (mi2oon) [inline app/templates/_webpack.config.js]: Love the use of the spread operator 👍

File: app/templates/_webpack.config.js (hunk 1 above)
Comment (mi2oon) [reply app/templates/_webpack.config.js]: Good one

File: app/templates/_webpack.config.js (hunk 3 above)
Comment (mi2oon) [reply app/templates/_webpack.config.js]: True, I noticed that for Vue.js but can't we just use `new webpack.EnvironmentPlugin(['NODE_ENV'])` in pulgins?

**Comment by racse1** [comment]
@mi2oon Forgot to mention, I also changed the chunk's filename, it's `[name].[chunkhash].chunk.js` now. This way it's easier to understand what's kind of file it is while you debugging in Chrome 😄 Does it sound reasonable?

**Pull request merged**